src/PKSim.CLI.Core/Services/SnapshotRunner.cs

```diff
@@ -98,28 +98,42 @@ private async Task startSnapshotRun(IReadOnlyList<FileMap> fileMaps, Func<FileMa
          _logger.AddInfo($"{fileMaps.Count} {"project".PluralizeIf(fileMaps)} loaded and exported in {timeSpent.ToDisplay()}");
       }
 
-      private async Task createProjectFromSnapshotFile(FileMap file)
+      private async Task createProjectFromSnapshotFile(FileMap file, bool runSimulations)
       {
          _logger.AddInfo($"Starting project export for '{file.SnapshotFile}'");
-         var project = await _snapshotTask.LoadProjectFromSnapshotFileAsync(file.SnapshotFile);
+         var project = await _snapshotTask.LoadProjectFromSnapshotFileAsync(file.SnapshotFile, runSimulations);
          if (project == null)
             return;
 
          _logger.AddDebug($"Snapshot loaded successfully from '{file.SnapshotFile}'");
          _workspace.Project = project;
-         _workspacePersistor.SaveSession(_workspace, file.ProjectFile);
+         try
+         {
+            _workspacePersistor.SaveSession(_workspace, file.ProjectFile);
+         }
+         finally
+         {
+            _workspacePersistor.CloseSession();
```

Comment:
@rwmcintosh Interesting. Where does this come from?

Reply:
I'll let @rwmcintosh provide more details on this but to summarise:
When we used this SnapshotRunner from R, the process was persisting after conversion was done.
File handle was also persisting, resulting in a bug when we tried to run several conversions in a row.